.github/workflows/ci.yml

```
@@ -109,12 +109,14 @@ jobs:
           - "3.9"
           - "3.10"
           - "3.11"
+          - key: "3.12"
+            full: "3.12.0-beta.3"
 
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-python@v4
         with:
-          python-version: ${{ matrix.python }}
+          python-version: ${{ matrix.python.full || matrix.python }}
```

you can use `allow-prereleases: true` if you don't care about the specific beta version to test.

c.f. https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#allow-pre-releases

---

```suggestion
          python-version: ${{ matrix.python }}-dev
```

This will gracefully move to using stable releases, once they are out.

---

Both these solutions would cause prereleases for other versions to be picked up. I’m not sure I’m on board with that.

---

> Both these solutions would cause prereleases for other versions to be picked up.

They won't have any though?

---

Will they not? I am not familiar with how GitHub publishes versions, but Python bug fixes do have prereleases officially.